src/coreclr/vm/threadsuspend.cpp

```diff
@@ -4600,6 +4605,9 @@ void Thread::HijackThread(ExecutionState *esb X86_ARG(ReturnKind returnKind) X86
 
     // Remember the place that the return would have gone
     m_pvHJRetAddr = *esb->m_ppvRetAddrPtr;
+#if defined(TARGET_ARM64)
+    m_pvHJRetAddr = PacStripPtr(m_pvHJRetAddr);
```

Comment:
Would it make sense to keep the bits on `m_pvHJRetAddr` unstripped so that the original return address is protected at all times? I think stripping the bits here creates a hole in the protection. Should we strip the bits later when we read `m_pvHJRetAddr` for stackwalking, but not when we use it to store/restore the original return address so that the return address is protected for the whole time?

Reply:
It's a good idea. Ideally, we should authenticate instead of stripping to ensure that we are returning to a valid return address. I'll check how can we do this.

Reply:
> the original address is preserved so that after GC we can return to it, and the updated address would take the execution flow to desired new address

Right. My point is that it would be better to keep the signed address intact through the whole process:
- Delete PacStripPtr in Thread::HijackThread and store the original signed address in m_pvHJRetAddr instead 
- Delete PacSignPtr in Thread::UnhijackThread
- Add PacStripPtr as necessary to places that read m_pvHJRetAddr and do not expect the signature in the upper bits

It will make the original return address protected while it is stored in m_pvHJRetAddr. As implemented currently, the return address is not protected while it is stored in m_pvHJRetAddr.